modules/tinymce/src/core/test/ts/browser/textpatterns/TextPatternsPublicApiTest.ts

```diff
@@ -0,0 +1,124 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Arr } from '@ephox/katamari';
+import { TinyAssertions, TinyHooks, TinyUiActions } from '@ephox/wrap-mcagar';
+
+// eslint-disable-next-line @tinymce/no-publicapi-module-imports
+import { Editor, PluginManager, TextPatterns } from 'tinymce/core/api/PublicApi';
+
+// The purpose of this test is to ensure that we have all the types required
+// for managing TextPatterns exposed through PublicApi
+describe('browser.tinymce.textpatterns.TextPatternsPublicApiTest', () => {
```

Comment:
I this the first time we have a test that only has the purpose of testing the types. Should these be a new category of tests or should type testing be else where?

Reply:
It's actually been done before fwiw. Here's the first one I can think of: https://github.com/tinymce/tinymce/blob/b69c943324af5e127fe53f296f8ff1d4df0769b2/modules/katamari/src/test/ts/atomic/api/data/CovarianceAssertionTest.ts.

Reply:
I think testing the presence of types on the PublicApi in lots of places sounds like a good way to make sure we don't accidentally break something. Especially when some of the exports from PublicApi are "all the exports in this other module"

Reply:
As long as refactoring doesn't auto-update the tests and make it pass incorrectly 😆

Reply:
Well, then it's the review's job to see that the test changed :P